### PR TITLE
Updated Sphinx configuration to 2.0.1 #2468

### LINKED ISSUE #2468: Additional broken links in documentation.
Similar to #2302 the link to the l2t style guide on  https://plaso.readthedocs.io/en/latest/sources/developer/Style-guide.html is broken.

The packaged release link here: https://plaso.readthedocs.io/en/latest/sources/user/Users-Guide.html#installing-the-packaged-release is also broken.

### PR DESCRIPTION
**Pull request by Onager**
Fixes #2468

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -396,48 +385,35 @@ def find_and_replace(self, node):
...
+      if (
+          reference_uri.startswith(l2tdocs_prefix) and
+          not reference_uri.endswith('.asciidoc')):
+        node['refuri'] = reference_uri + '.md'
+        print('{0:s} -> {1:s}'.format(reference_uri, node['refuri']))
     return node
 
-  def traverse(self, node):
+  def _Traverse(self, node):
     """Traverse the document tree rooted at node.
```
Comment (joachimmetz) [inline docs/conf.py]: Nit "Traverse" => "Traverses"

File: docs/conf.py [hunk 2]
```diff
@@ -396,48 +385,35 @@ def find_and_replace(self, node):
...
 
     node(node): docutils node.
```
Comment (joachimmetz) [inline docs/conf.py]: add `Args:`

File: docs/conf.py [hunk 3]
```diff
@@ -372,21 +359,23 @@ def __add__(self, other):
...
 
   default_priority = 1000
 
   ANCHOR_REGEX = re.compile(
       r'(?P<uri>[a-zA-Z0-9-./]+?).md#(?P<anchor>[a-zA-Z0-9-]+)')
 
-  def find_and_replace(self, node):
-    """Parses URIs containing .md and replaces them with their HTML page.
+  def _FixLinks(self, node):
+    """Correct links to .md files hosted on l2tdocs.
```
Comment (joachimmetz) [inline docs/conf.py]: nit "Correct" => "Corrects"

File: docs/conf.py [hunk 4]
```diff
@@ -372,21 +359,23 @@ def __add__(self, other):
...
 
     Args:
         node(node): docutils node.
```
Comment (joachimmetz) [inline docs/conf.py]: nit only 2 addition spaces indentation

File: docs/conf.py [hunk 5]
```diff
@@ -372,21 +359,23 @@ def __add__(self, other):
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 # texinfo_no_detailmenu = False
 
+def RunSphinxAPIDoc(_):
```
Comment (joachimmetz) [inline docs/conf.py]: replace `_` by a name with meaning.

File: docs/conf.py [hunk 6]
```diff
@@ -396,48 +385,35 @@ def find_and_replace(self, node):
...
-  def traverse(self, node):
+  def _Traverse(self, node):
     """Traverse the document tree rooted at node.
 
     node(node): docutils node.
     """
-    self.find_and_replace(node)
+    self._FixLinks(node)
 
     for c in node.children:
```
Comment (joachimmetz) [inline docs/conf.py]: c => child_node or sub_node (or equiv)

File: docs/conf.py (hunk 2 above)
Comment (joachimmetz) [reply docs/conf.py]: Also use "[...]" => "node [node]: ..."

Comment (joachimmetz) [reply docs/conf.py]: Also node looks like a dict? or is it sphinx.node?

File: docs/conf.py [hunk 7]
```diff
@@ -372,21 +359,23 @@ def __add__(self, other):
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 # texinfo_no_detailmenu = False
 
+def RunSphinxAPIDoc(_):
+  """Runs sphinx-apidoc to auto-generate documentation."""
+  current_directory = os.path.abspath(os.path.dirname(__file__))
+  module = os.path.join(current_directory, '..', 'plaso')
```
Comment (joachimmetz) [inline docs/conf.py]: module => module_path or module_directory

also maybe use os.path.dirname() instead of '..'

File: docs/conf.py (hunk 5 above)
Comment (Onager) [reply docs/conf.py]: Done

File: docs/conf.py (hunk 7 above)
Comment (Onager) [reply docs/conf.py]: Done

File: docs/conf.py (hunk 3 above)
Comment (Onager) [reply docs/conf.py]: Done

File: docs/conf.py (hunk 4 above)
Comment (Onager) [reply docs/conf.py]: Done

File: docs/conf.py (hunk 1 above)
Comment (Onager) [reply docs/conf.py]: Done

File: docs/conf.py (hunk 6 above)
Comment (Onager) [reply docs/conf.py]: Done.

File: docs/conf.py (hunk 2 above)
Comment (Onager) [reply docs/conf.py]: Done.

File: docs/conf.py [hunk 8]
```diff
@@ -129,8 +127,6 @@ def __add__(self, other):
 modules_to_mock = set(modules_to_mock).difference(PIP_INSTALLED_MODULES)
 modules_to_mock = sorted(modules_to_mock)
 print('Mocking modules')
```
Comment (joachimmetz) [inline docs/conf.py]: remove print?

File: docs/conf.py [hunk 9]
```diff
@@ -372,72 +359,73 @@ def __add__(self, other):
...
+# here: https://www.sphinx-doc.org/en/master/extdev/appapi.html#events
+def RunSphinxAPIDoc(unused_app):
+  """Runs sphinx-apidoc to auto-generate documentation.
 
-# Configure sphinx to convert markdown links (recommonmark is broken at the
-# moment).
-
+  Args:
+    unused_app (sphinx.application.Sphinx): Sphinx application. Required by the
+      the Sphinx event callback API.
```
Comment (joachimmetz) [inline docs/conf.py]: `+` 2 indentation spaces

**Comment by joachimmetz** [comment]
@Onager please have a look at the remaining nits, and merge once they've been addressed

**Pull request merged**